src/main/java/com/hivemq/cli/commands/shell/ListClientsCommand.java

```diff
@@ -137,9 +134,9 @@ public void run() {
 
             for (final String key : sortedKeys) {
 
-                final MqttClient client = cache.get(key);
+                final MqttClient client = clientKeysToClientData.get(key).getClient();
 
-                final LocalDateTime dateTime = clientDataMap.get(key).getCreationTime();
+                final LocalDateTime dateTime = clientKeysToClientData.get(key).getCreationTime();
```

Comment:
clientKeysToClientData.get(key) can be extracted into a local variable, so the map query only needs to be done once.